modules/ding_facetbrowser/ding_facetbrowser.module

```diff
@@ -112,14 +113,138 @@ function ding_facetbrowser_libraries_info() {
 }
 
 /**
- * Create a custom form with checkbox elements for facets and ajax callback
+ * Get the year_span element.
+ *
+ * @param string $search
+ *   The searchkey.
+ * @param array $conditions
+ *   List of conditions.
+ *
+ * @return array
+ *   A year span form element container.
+ */
+function _ding_facetbrowser_form_year_span_element($search, array $conditions) {
+  // To get the correct counter for the span, we need to run the
+  // search without the actual date filter on, so the complete dataset
+  // is included.
+  $count_search = array(
+    'run' => FALSE,
+    'conditions' => $conditions,
+    'searchkey' => $search,
+  );
+  $range = array();
+
+  // Remove all facet.date conditions.
+  if (!empty($count_search['conditions']['facets'])) {
+    foreach ($count_search['conditions']['facets'] as $delta => $condition_facet) {
+      if (strpos($condition_facet, 'facet.date:') === 0) {
+        $parts = explode(':', $condition_facet);
+
+        // Set the range to match the year span.
+        if (isset($parts[2])) {
+          switch ($parts[2]) {
+            case '>=';
+              $range[0] = (int) $parts[1];
+              break;
+
+            case '<=';
+              $range[1] = (int) $parts[1];
+              break;
+          }
+        }
+
+        $count_search['run'] = TRUE;
+        unset($count_search['conditions']['facets'][$delta]);
+      }
+    }
+  }
+
+  // Only run the counter, if it's different from the search already
+  // performed.
+  if ($count_search['run'] === TRUE) {
+    $count_search['conditions']['result_id'] = 'ding_facetbrowser_count_search';
+    search_data($count_search['searchkey'], 'ting_search', $count_search['conditions']);
```

Comment:
From a performance perspective it is unfortunate that this issues a new search - especially if the search engine is slow.

I understand that we want to use the full spectrum. One suggestion could be to store the minimum and maximum values as hidden entries in the form if such values have not been set already. This would ensure that the values persist between requests without having to rerun the search.